common/src/main/java/com/wynntils/services/mapdata/providers/json/JsonMapAttributesBuilder.java

```diff
@@ -63,8 +68,55 @@ public JsonMapAttributesBuilder setIconVisibility(JsonMapVisibility iconVisibili
         return this;
     }
 
-    public JsonMapAttributes build() {
-        return new JsonMapAttributes(
-                label, icon, priority, level, labelColor, labelShadow, labelVisibility, iconColor, iconVisibility);
+    public JsonMapLocationAttributesBuilder asLocationAttributes() {
+        return new JsonMapLocationAttributesBuilder();
+    }
+
+    public JsonMapPathAttributesBuilder asPathAttributes() {
+        return new JsonMapPathAttributesBuilder();
+    }
+
+    public JsonMapAreaAttributesBuilder asAreaAttributes() {
+        return new JsonMapAreaAttributesBuilder();
+    }
+
+    protected void checkInvalidAttribute(String fieldName) {
+        if (!WynntilsMod.isDevelopmentBuild()) return;
```

Comment:
The PR description mentions that we check invalid runtime-built attributes in development builds too, but this is Json. Shouldn't we change up the logic here to check this in non-development environments too? Or is it pending the error propagation, in which case, a FIXME would suffice here.

Reply:
Sorry, was a bit tired and sleepy when I published the PR. What I meant that we have two sources for mapdata:
1) From Java code (built in only). These are checked by the Java inheritance stuff.
2) From json. These are checked by this code.

Reply:
That still means json error propagation could be done sometime after this PR, but I see why it's only done in development environments. I think we can merge this PR now.

Reply:
Ok, confusion galore here. This is something I want to address but thought I'd do after this change is integrated into mapdata.

This file has a misleading name. We have several `Json` files in mapdata that is not really json-related; what it means is that it is just a simple concrete class that holds variables for all attributes. Sure, such a class is created by GSON, and it was probably the first ever use of that class (hence the name) but now it has slipped out and is all over the place. So the "json builder" is in fact not at all used when reading json files, but when constructing java objects for built-in providers.

Reply:
I suppose that warrants a change in terminology around the class then.